combustion-android-ble/src/main/java/inc/combustion/framework/ble/device/NodeBleDevice.kt

```diff
@@ -23,34 +27,62 @@ internal class NodeBleDevice(
     private var uart: UartBleDevice = UartBleDevice(mac, nodeAdvertisingData, owner, adapter)
 ) {
 
-    protected val genericRequestHandler = UartBleDevice.MessageCompletionHandler()
-    protected val readFeatureFlagsRequest = UartBleDevice.MessageCompletionHandler()
+    private val genericRequestHandler = UartBleDevice.MessageCompletionHandler()
+    private val readFeatureFlagsRequest = UartBleDevice.MessageCompletionHandler()
 
     companion object {
-        const val NODE_MESSAGE_RESPONSE_TIMEOUT_MS = 5000L
+        const val NODE_MESSAGE_RESPONSE_TIMEOUT_MS = 30000L
     }
 
     init {
         processUartMessages()
     }
 
-    val rssi: Int get() { return uart.rssi }
-    val connectionState: DeviceConnectionState get() { return uart.connectionState }
-    val isConnected: Boolean get() { return uart.isConnected.get() }
-    val isDisconnected: Boolean get() { return uart.isDisconnected.get() }
-    val isInRange: Boolean get() { return uart.isInRange.get() }
-    val isConnectable: Boolean get() { return uart.isConnectable.get() }
+    val rssi: Int
+        get() {
+            return uart.rssi
+        }
+    val connectionState: DeviceConnectionState
+        get() {
+            return uart.connectionState
+        }
+    val isConnected: Boolean
+        get() {
+            return uart.isConnected.get()
+        }
+    val isDisconnected: Boolean
+        get() {
+            return uart.isDisconnected.get()
+        }
+    val isInRange: Boolean
+        get() {
+            return uart.isInRange.get()
+        }
+    val isConnectable: Boolean
+        get() {
+            return uart.isConnectable.get()
+        }
 
-    val deviceInfoSerialNumber: String? get() { return uart.serialNumber }
+    val deviceInfoSerialNumber: String?
+        get() {
+            return uart.serialNumber
+        }
 
+    @Synchronized
```

Comment:
What was the problem was happening when this was called by different threads?

The call to `genericRequestHandler.wait` sets variables that are reused/reset with the next call to `sendNodeRequest`. Does this prevent that from happening?

Reply:
Ah, you have a point - I made some other change and that actually fixed the issue and I just assumed... Let me see if there is an easy equivalent fix or I will just remove.

Reply:
The problem is that calls get stuck with no response callbacks happening.

Reply:
Switched to a `Semaphore` solution in `NetworkManager`.